lib/set-script.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const log = require('npmlog')
+const usageUtil = require('./utils/usage.js')
+const { localPrefix } = require('./npm.js')
+const fs = require('fs')
+const usage = usageUtil('set-script', 'npm set-script [<script>] [<command>]')
+const completion = require('./utils/completion/none.js')
+
+const cmd = (args, cb) => set(args).then(() => cb()).catch(cb)
+
+const set = async function(args) {
+    // Parse arguments
+    if (args.length !== 2) {
+        log.error('set-script', 'Expected 2 arguments: got ' + args.length)
+        return
+    }
+
+    // Set the script
+    let manifest
+    let warn = false
+    try {
+        manifest = fs.readFileSync(localPrefix + '/package.json', 'utf-8')
+    } catch {
+        log.error('set-script', 'package.json not found')
+        return
+    }
+    try {
+        manifest = JSON.parse(manifest)
+    } catch(error) {
+        log.error('set-script', 'Invalid package.json: ' + error)
+        return
+    }
+    if (!manifest.scripts) {
+        manifest.scripts = {}
+    }
+    if (manifest.scripts[args[0]] && manifest.scripts[args[0]] !== args[1]) {
+        warn = true
+    }
+    manifest.scripts[args[0]] = args[1]
+    manifest = JSON.stringify(manifest, null, 2) + '\n'
```

Comment:
Here you're going to need to use `read-package-json-fast` and a bit more elaborated logic in order to keep indentation and respect line breaks from the original file, again, make sure to require the module:

```js
const rpj = require('read-package-json-fast')
```

And then you can do something like:

```suggestion
  // format content
  const packageJsonInfo = await rpj('localPrefix + '/package.json')
  const {
    [Symbol.for('indent')]: indent,
    [Symbol.for('newline')]: newline,
  } = packageJsonInfo
  const format = indent === undefined ? '  ' : indent
  const eol = newline === undefined ? '\n' : newline
  const content = (JSON.stringify(manifest, null, format) + '\n')
    .replace(/\n/g, eol)

```

In order to store it back with the same format 😊